hedera-node/docs/design/services/smart-contract-service/hbar-allowance-approval.md

```diff
@@ -0,0 +1,73 @@
+# hbar Allowance and Approval
+
+## Purpose
+
+Currently, the IHederaTokenService interface in the Hedera Smart Contract Service offers support for approving an allowance to a `spender` account for tokens.
+Document the implementation of the hbar allowance and approval functionality in the Hedera Smart Contract Service which extends this functionality to an account's hbars.
+
+## Goals
+
+Define the classes necessary to complete implementation of the hbar allowance and approval functionality in the Hedera Smart Contract Service.  Only the implementation in the modularized code will be described in this document.
+
+## Non Goals
+
+The implementation for crypto allowance and approval in the Token Service Module will only be described at a high level as this implementation is already completed.
+
+## Architecture
+
+In order to support both EOA and contract callers to these functions, a new interface `IHRC<###>` will be supported and implemented using the pattern established for token proxy contracts as discussed in the `Specification` section of [HIP-719](https://hips.hedera.com/hip/hip-719). 
+The difference is that `IHRC[###]` will act on an account address rather than a token address will be the account in view with respect to the allowance to the `spender` account for the specified amount.  A new proxy contract and a new system contract to handle these transaction requests will be implemented.
```

Comment:
```suggestion
The difference is that `IHRC[###]` will act on an account address rather than a token address will be the account in view with respect to the allowance to the `spender` account for the specified amount.  A new proxy contract and a new system contract to handle these transaction requests will be implemented similar to HIP 632 logic.
```

Reply:
Done in new pr for hbar approval